src/components/Form/Checkbox.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { Component, ReactNode } from 'react';
+import classnames from 'classnames';
+import { HtmlInput, HtmlLabel, HtmlSpan, HtmlDiv } from '../../reset';
+
+import { idGenerator } from '../../utils/uuid';
+
+const baseClassName = 'fi-checkbox';
+
+export const checkboxBaseClassNames = {
+  disabled: `${baseClassName}--disabled`,
+  container: `${baseClassName}_container`,
+  input: `${baseClassName}_input`,
+  label: `${baseClassName}_label`,
+  statusText: `${baseClassName}_status`,
+  hintText: `${baseClassName}_hintText`,
+};
+
+type CheckboxVariant = 'small' | 'large';
+type CheckboxStatus = 'default' | 'error' | 'disabled';
+
+export interface CheckboxInputProps {
+  /** State of input checkbox */
+  checked?: boolean;
+  /** Custom classname for the container */
+  className?: string;
+  disabled?: boolean;
+  'aria-label'?: string;
+  'aria-labelledby'?: string;
+}
+
+export interface CheckboxProps {
+  /** Controlled checked-state - user actions use onClick to change  */
+  checked?: boolean;
+  /** Default status of Checkbox when not using controlled 'checked' state
+   * @default false
+   */
+  defaultChecked?: boolean;
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Disable Checkbox. Value won't be included when submitting */
+  disabled?: boolean;
+  /** Event handler to execute when clicked */
+  onClick?: ({ checkboxState }: { checkboxState: boolean }) => void;
+  /**
+   * Label element content
+   */
+  children?: ReactNode;
+  /**
+   * 'small' | 'large'
+   * @default small
+   */
+  variant?: CheckboxVariant;
+  /**
+   * 'default' | 'error' | 'disabled'
+   * @default default
+   */
+  status?: CheckboxStatus;
+  /**
+   * Status text to be displayed in the status text element.
+   */
+  statusText?: string;
+  /**
+   * Hint text to be displayed under the label.
+   */
+  hintText?: string;
+  /** Pass custom props to Checkbox's input component/element */
+  checkboxInputProps?: CheckboxInputProps;
+  /**
+   * aria-label for the HTML input-element,
+   * alternatively you can define aria-labelledby with label-element id
+   */
+  'aria-label'?: string;
+  'aria-labelledby'?: string;
+  /** Unique id
+   * @default uuidV4
+   */
+  id?: string;
+}
+
+export interface CheckboxState {
+  checkboxState: boolean;
+}
+
+export class Checkbox extends Component<CheckboxProps> {
+  state = {
+    checkboxState: !!this.props.checked || !!this.props.defaultChecked || false,
+  };
+
+  /**
+   * This is deprecated and should be replaced
+   */
+  componentWillReceiveProps(nextProps: CheckboxProps) {
```

Comment:
As you have commented already about the deprecation, but pasting the React's documentation about this for explanation:
https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops

Reply:
This approach needs to be changed. Will be depricated as of React 17 and name should be UNSAFE_componentWillReceiveProps as we use React 16.9 which deprecates the original name. In addition the documentation behind the link above advices to use a different approach instead of using this lifecycle method. I propose to find an alternative method.

Reply:
This should be doable with getDerivedStateFromProps using similar approach with Toggle and Expander.

Reply:
This seems to be fixed as well.